packages/machine/src/models/state-channel.ts

```diff
@@ -171,9 +183,11 @@ export class StateChannel {
     multisigAddress: string,
     multisigOwners: string[]
   ) {
+    const sortedMultisigOwners = sortAddresses(multisigOwners);
```

Comment:
To remain consistent with the constructor, this should also throw if the given `multisigOwners` is not sorted.

On a side note: setting up a channel is always going to be the first step because the free balance needs to be setup, which involves the `NetworkContext` needing to be passed in, so why not require the constructor to have the network context as a parameter and do the setup in the constructor?

Needing to instantiate a new state channel object then calling setup on it seems like a sub-optimal creation pattern if it can be avoided.

Reply:
`setupChannel` has been changed to a static function

Reply:
So _only_ `setupChannel` should be called then? and the constructor is hidden away?

That's what it sounds like

Reply:
The constructor is actually publicly exposed but it is not called publicly

Reply:
I think there are cases where we may want a `StateChannel` that has not been set up yet.